### PR TITLE
[tests-only][full-ci]add api test for search files using tag in project spaces

### PR DESCRIPTION
**Pull request by nabim777**
## Description
This PR adds the API tests for full search api. The scenarios added in this PR are
- search files by tag using new DAV path in Project spaces
In this PR there is refactored on:
- search files by tag

## Related Issue
- https://github.com/owncloud/ocis/issues/6606#event-9635806756

## Motivation and Context
- there was no test coverage for api test for search files using tag in project spaces. This PR covers the global search using tag on  personal space and global spaces. There is only search scanerio test done only by using new Dav path. There is also done refactor on `search files by tag` there are some changes like:
1. The file name is changes
2. There is also check for search for file inside folder and subfolder

## How Has This Been Tested?
- test environment:
- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiGraph/fullSearch.feature [hunk 1]
```diff
@@ -8,27 +8,70 @@ Feature: full text search
...
     And the search result of user "Alice" should contain only these files:
-      | file1.txt |
-      | file2.txt |
+      | test.txt                |
+      | fileInsideFolder.txt    |
+      | fileInsideSubFolder.txt |
     Examples:
       | dav-path-version |
       | old              |
-      | new              |
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/fullSearch.feature:28]: why was this removed for this scenario?

File: tests/acceptance/features/apiGraph/fullSearch.feature [hunk 2]
```diff
@@ -8,27 +8,70 @@ Feature: full text search
...
       | spaces           |
 
 
+  Scenario: search files by tag using new DAV path
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/fullSearch.feature:36]: what about old and spaces endpoints?

Comment (nabim777) [reply tests/acceptance/features/apiGraph/fullSearch.feature:36]: while searching in old and spaces endpoint the results entries cames zero

File: tests/acceptance/features/apiGraph/fullSearch.feature (hunk 1 above)
Comment (nabim777) [reply tests/acceptance/features/apiGraph/fullSearch.feature:28]: sorry oh I forget that

File: tests/acceptance/features/apiGraph/fullSearch.feature [hunk 3]
```diff
@@ -8,27 +8,75 @@ Feature: full text search
...
+      | fileInsideFolder.txt    |
+      | fileInsideSubFolder.txt |
     Examples:
       | dav-path-version |
       | old              |
       | new              |
       | spaces           |
 
 
+  Scenario Outline: search files by tag using new DAV path
```
Comment (SagarGi) [inline tests/acceptance/features/apiGraph/fullSearch.feature:37]: using new dav but i see old as well?

File: tests/acceptance/features/apiGraph/fullSearch.feature [hunk 4]
```diff
@@ -8,27 +8,75 @@ Feature: full text search
...
+      | personalFile.txt                |
+      | personalFileInsideFolder.txt    |
+      | personalFileInsideSubFolder.txt |
+      | spacesFile.txt                  |
+      | spacesFileInsideFolder.txt      |
+      | spacesFileInsideSubFolder.txt   |
+    Examples:
+      | dav-path-version |
+      | old              |
+      | new              |
```
Comment (SagarGi) [inline tests/acceptance/features/apiGraph/fullSearch.feature:77]: what about spaces web dav? it should work right?

File: tests/acceptance/features/apiGraph/fullSearch.feature (hunk 3 above)
Comment (SwikritiT) [inline tests/acceptance/features/apiGraph/fullSearch.feature:37]: this scenario is too long. is it possible to shorten it? or split it into various scenarios for personal, project space and subfolders?

File: tests/acceptance/features/apiGraph/fullSearch.feature [hunk 5]
```diff
@@ -8,27 +8,75 @@ Feature: full text search
...
+    And user "Alice" has created folder "folderWithFile/subFolder/"
+    And user "Alice" has uploaded file with content "hello world" to "test.txt"
+    And user "Alice" has uploaded file with content "Namaste nepal" to "folderWithFile/fileInsideFolder.txt"
+    And user "Alice" has uploaded file with content "hello nepal" to "folderWithFile/subFolder/fileInsideSubFolder.txt"
+    And user "Alice" has created the following tags for file "test.txt" of the space "Personal":
       | tag1 |
-    And user "Alice" has created the following tags for file "file2.txt" of the space "Personal":
+    And user "Alice" has created the following tags for file "folderWithFile/fileInsideFolder.txt" of the space "Personal":
+      | tag1 |
+    And user "Alice" has created the following tags for file "folderWithFile/subFolder/fileInsideSubFolder.txt" of the space "Personal":
```
[the remark this answers is not in the document]
Comment (saw-jan) [reply tests/acceptance/features/apiGraph/fullSearch.feature:22]: Idk but I think good to have it tested

File: tests/acceptance/features/apiGraph/fullSearch.feature (hunk 3 above)
Comment (saw-jan) [reply tests/acceptance/features/apiGraph/fullSearch.feature:37]: we have steps like:
`has created the following folders`
`has uploaded the following files with content`
Maybe that will help

Comment (saw-jan) [reply tests/acceptance/features/apiGraph/fullSearch.feature:37]: but probably not for project space

Comment (nabim777) [reply tests/acceptance/features/apiGraph/fullSearch.feature:41]: Oh, I forget to change the description of the scenario.
Initially, the old Davpath was not working but after some changes in code, it works.

File: tests/acceptance/features/apiGraph/fullSearch.feature (hunk 4 above)
Comment (nabim777) [reply tests/acceptance/features/apiGraph/fullSearch.feature:85]: While searching files in project space using spaces davpath the result came zero result as like in searching the share files using spaces dav path. Issues like this https://github.com/owncloud/ocis/issues/6669

File: tests/acceptance/features/apiGraph/fullSearch.feature [hunk 6]
```diff
@@ -8,27 +8,70 @@ Feature: full text search
...
 
 
-  Scenario Outline: search files using a tag
+  Scenario Outline: search files by tag
     Given using <dav-path-version> DAV path
-    And user "Alice" has uploaded file with content "hello world" to "file1.txt"
-    And user "Alice" has uploaded file with content "Namaste nepal" to "file2.txt"
-    And user "Alice" has uploaded file with content "hello nepal" to "file3.txt"
-    And user "Alice" has created the following tags for file "file1.txt" of the space "Personal":
+    And user "Alice" has created folder "folderWithFile"
```
[the remark this answers is not in the document]
Comment (nabim777) [reply tests/acceptance/features/apiGraph/fullSearch.feature]: ok

File: tests/acceptance/features/apiGraph/fullSearch.feature [hunk 7]
```diff
@@ -8,27 +8,64 @@ Feature: full text search
...
       | spaces           |
 
 
+  Scenario Outline: search project space files by tag
+    Given using spaces DAV path
+    And the administrator has assigned the role "Space Admin" to user "Alice" using the Graph API
+    And user "Alice" has created a space "tag-space" with the default quota using the GraphApi
+    And user "Alice" has created a folder "spacesFolderWithFile/spacesSubFolder" in space "tag-space"
+    And user "Alice" has uploaded a file inside space "tag-space" with content "tagged file" to "spacesFile.txt"
+    And user "Alice" has uploaded a file inside space "tag-space" with content "non-tagged file" to "spacesFileWithoutTag.txt"
```
Comment (SwikritiT) [inline tests/acceptance/features/apiGraph/fullSearch.feature:47]: maybe this?
```suggestion
    And user "Alice" has uploaded a file inside space "tag-space" with content "untagged file" to "spacesFileWithoutTag.txt"
```

File: tests/acceptance/features/apiGraph/fullSearch.feature (hunk 4 above)
Comment (SwikritiT) [reply tests/acceptance/features/apiGraph/fullSearch.feature:66]: is there a issue for the project spaces too? if your sure it doesn't work please create a seperate issue for project sapces too or update the mentioned issue to include that the behaviour is same for project space as well.

File: tests/TestHelpers/WebDavHelper.php [hunk 8]
```diff
@@ -747,7 +747,7 @@ public static function getDavPath(
 			if ($type === "trash-bin") {
 				return "/remote.php/dav/spaces/trash-bin/" . $spaceId . '/';
 			}
-			return "dav/spaces/" . $spaceId . '/';
+			return "/remote.php/dav/spaces/" . $spaceId . '/';
```
Comment (saw-jan) [inline tests/TestHelpers/WebDavHelper.php:750]: @nabim777 link to the issue?

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 9]
```diff
@@ -259,6 +259,7 @@ public function usingOldDavPath():void {
 		$this->davPath = $this->getOldDavPath();
 		$this->usingOldDavPath = true;
 		$this->customDavPath = null;
+		$this->usingSpacesDavPath = false;
```
Comment (amrita-shrestha) [inline tests/acceptance/features/bootstrap/WebDav.php:262]: do we need this?

File: tests/TestHelpers/WebDavHelper.php (hunk 8 above)
Comment (nabim777) [reply tests/TestHelpers/WebDavHelper.php:750]: https://github.com/owncloud/ocis/issues/6954

File: tests/acceptance/features/bootstrap/WebDav.php (hunk 9 above)
Comment (saw-jan) [reply tests/acceptance/features/bootstrap/WebDav.php:262]: similar to other functions. I think it got missed out here.

File: tests/TestHelpers/WebDavHelper.php (hunk 8 above)
Comment (saw-jan) [reply tests/TestHelpers/WebDavHelper.php:750]: https://github.com/owncloud/ocis/issues/6954#issuecomment-1663766325

File: tests/acceptance/features/apiGraph/fullSearch.feature [hunk 10]
```diff
@@ -8,20 +8,58 @@ Feature: full text search
...
-    And user "Alice" has uploaded file with content "Namaste nepal" to "file2.txt"
-    And user "Alice" has uploaded file with content "hello nepal" to "file3.txt"
-    And user "Alice" has created the following tags for file "file1.txt" of the space "Personal":
+    And user "Alice" has created the following folders
+      | path                      |
+      | folderWithFile            |
+      | folderWithFile/subFolder/ |
+    And user "Alice" has uploaded the following files with content "some data"
+      | path                                             |
+      | test.txt                                         |
```
Comment (SwikritiT) [inline tests/acceptance/features/apiGraph/fullSearch.feature:19]: waht about this?
```suggestion
      | fileInRootLevel.txt                                         |
```

Comment (nabim777) [reply tests/acceptance/features/apiGraph/fullSearch.feature]: Of course, determining if a file is at the root level will be simple.

**Review by saw-jan — COMMENTED** [review]
LGTM

**Review by SwikritiT — APPROVED** [review]
LGTM

**Review by amrita-shrestha — APPROVED** [review]
lgtm
